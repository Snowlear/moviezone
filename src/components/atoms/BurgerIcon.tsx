import BurgerStyles from "./BurgerIcon.module.css";

export default function BurgerIcon({ onClick, ishiddenMenu }: ContainerProps) {
  return (
    <i
      onClick={() => onClick()}
      className={
        BurgerStyles.burgerMenu +
        " " +
        (ishiddenMenu ? "fas fa-bars" : "fas fa-times")
      }
    ></i>
  );
}

interface ContainerProps {
  onClick: Function;
  ishiddenMenu: boolean;
}
