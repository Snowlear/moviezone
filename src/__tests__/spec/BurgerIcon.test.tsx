import React from "react";
import renderer from 'react-test-renderer';
import BurgerIcon from "../../components/atoms/BurgerIcon";

describe("<Box />", () => {
  it("renders <Box /> component with data.", async () => {
    const tree = renderer
    .create(<BurgerIcon ishiddenMenu={true} onClick={jest.fn()}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});