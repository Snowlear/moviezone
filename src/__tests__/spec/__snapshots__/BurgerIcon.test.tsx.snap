// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders <Box /> component with data. 1`] = `
<i
  className="undefined fas fa-bars"
  onClick={[Function]}
/>
`;
